api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -136,6 +136,21 @@ public void saveEncounter_shouldUpdateExistingEncounterWhenAChildObsIsEdited() t
 	}
 
 	@Test
+	@Verifies(value = "should update leaf obs and not duplicate at encounter level", method = "saveEncounter(Encounter)")
+	public void saveEncounter_shouldUpdateValueOfLeafObsAndNotDuplicateAtEncounterLevel() throws Exception {
+		executeDataSet(ENC_OBS_HIERARCHY_DATA_XML);
+		Encounter encounter = Context.getEncounterService().getEncounter(100);
+		assertEquals(1, encounter.getObsAtTopLevel(true).size());
+		Obs topLevelObs = encounter.getObsAtTopLevel(true).iterator().next();
+		topLevelObs.getGroupMembers().iterator().next().setValueText("editing first obs");
+		encounter.addObs(topLevelObs);
+
+		Encounter savedEncounter = Context.getEncounterService().saveEncounter(encounter);
+
+		assertEquals(1, savedEncounter.getObsAtTopLevel(true).size());
```

Comment:
Hmmm... So you're imagining a situation where say one calls saveEncounter() from within another transactional method and the session might not yet be flushed since the outer transaction is still going on, that might make sense I guess, okay we can keep the test